rotkehlchen/externalapis/etherscan.py

```diff
@@ -238,7 +239,15 @@ def _query(
         an unexpected response is returned. Also in the case of exhausting the backoff time.
         """
         result = None
-        query_str = f'https://{self.prefix_url}{self.base_url}/api?module={module}&action={action}'
+        cached_settings = CachedSettings()
+        if cached_settings.get_settings().use_unified_etherscan_api:
+            self.service_name = ExternalService.ETHERSCAN
+            query_str = f'https://api.etherscan.io/v2/api?chainid={self.chain.to_chain_id().serialize()}&'
```

Comment:
maybe you can save as an attribute this and save 2 function calls per etherscan call?